### PR TITLE
[staking miner]: add CLI tests + fix `--version`

### PR DESCRIPTION
**Pull request by niklasad1**

### CODE REVIEW

File: utils/staking-miner/src/main.rs [hunk 1]
```diff
@@ -317,40 +317,41 @@ struct MonitorConfig {
...
 
 	/// Force create a new snapshot, else expect one to exist onchain.
 	#[clap(long)]
 	force_snapshot: bool,
 }
 
-#[derive(Debug, Clone, Parser)]
-struct SharedConfig {
+#[derive(Debug, Clone, Parser, PartialEq)]
+#[clap(author, version, about)]
```
Comment (niklasad1) [inline utils/staking-miner/src/main.rs:353]: https://github.com/clap-rs/clap/blob/v3.1.5/examples/derive_ref/README.md#command-attributes

without `--version` here the version field is elided AFAIU

File: utils/staking-miner/src/main.rs [hunk 2]
```diff
@@ -317,40 +317,41 @@ struct MonitorConfig {
...
 	#[clap(subcommand)]
-	solver: Solvers,
+	solver: Solver,
 
 	/// The number of top backed winners to take. All are taken, if not provided.
 	take: Option<usize>,
 }
 
-#[derive(Debug, Clone, Parser)]
+#[derive(Debug, Clone, Parser, PartialEq)]
```
Comment (niklasad1) [inline utils/staking-miner/src/main.rs:337]: `PartialEq` needed for tests...

Comment (kianenigma) [inline utils/staking-miner/src/main.rs:337]:
```suggestion
#[derive(Debug, Clone, Parser)]
#[cfg_attr(test, Derive(PartialEq))]
```

**Review by kianenigma — APPROVED** [review]
awesome

File: utils/staking-miner/Cargo.toml [hunk 3]
```diff
@@ -44,4 +44,4 @@ westend-runtime = { path = "../../runtime/westend" }
 sub-tokens = { git = "https://github.com/paritytech/substrate-debug-kit", branch = "master" }
 
 [dev-dependencies]
-sp-version = { git = "https://github.com/paritytech/substrate", branch = "master" }
+assert_cmd = "2.0.2"
```
Comment (kianenigma) [inline utils/staking-miner/Cargo.toml:47]:
```suggestion
assert_cmd = "2.0.2"

```

File: utils/staking-miner/src/main.rs [hunk 4]
```diff
@@ -644,4 +644,102 @@ mod tests {
 		assert_eq!(polkadot_version.spec_name, "polkadot".into());
 		assert_eq!(kusama_version.spec_name, "kusama".into());
 	}
+
+	#[test]
+	fn cli_version_works() {
```
[the remark this answers is not in the document]
Comment (emostov) [reply utils/staking-miner/src/main.rs:649]: Ah I see its just version

**Review by emostov — APPROVED** [review]
LGTM - if its not too much of a hassle would be a plus to get the version test to work

File: utils/staking-miner/src/main.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (bkchr) [reply utils/staking-miner/src/main.rs:649]: The docs show that you need to write this as an integration test: https://docs.rs/assert_cmd/latest/assert_cmd/cargo/index.html#limitations

[the remark this answers is not in the document]
Comment (niklasad1) [reply utils/staking-miner/src/main.rs:649]: oh, I see ty

**Comment by niklasad1** [comment]
bot merge

**Pull request merged**